prow/plugins/verify-owners/verify-owners.go

```diff
@@ -205,9 +205,13 @@ func handle(ghc githubClient, gc git.ClientFactory, roc repoownersClient, log *l
 
 	// List modified OWNERS files.
 	var modifiedOwnersFiles []github.PullRequestChange
+	var ownerModified bool
 	for _, change := range changes {
-		if filepath.Base(change.Filename) == ownersFileName && change.Status != github.PullRequestFileRemoved {
-			modifiedOwnersFiles = append(modifiedOwnersFiles, change)
+		if filepath.Base(change.Filename) == ownersFileName {
```

Comment:
This doesn't change anything about he behavior of the plugin and does not address the issue mentioned in https://github.com/kubernetes/test-infra/issues/19607

The issue mentioned in https://github.com/kubernetes/test-infra/issues/19607 is that a PR at some point changed an OWNERS file and that change was invalid, the invalid owners label gets added. Later on the PR gets updated to not include any OWNERS file at all anymore, but the plugin does not remove that label.

What we need here is:
* A check if the invalid owners file exists
* Code to remove it in the `len(modifiedOwnersFiles) == 0 && !ownerAliasesModified {` condition if it exists

Reply:
Ah, ok. I interpreted this task in the wrong direction. Thanks for pointing out.
Now I updated code, to only short circuit if PR does not have invalid `do-not-merge/invalid-OWNERS-file` label.
```go
if len(modifiedOwnersFiles) == 0 && !ownerAliasesModified && !hasInvalidOwnersLabel {
	return nil
}
```
Decided don't duplicate the label removal code. Hope this will be correct solution.